### ISSUE TITLE
Conflicting definitions are in scope for kind `Effect` from the following modules

### ISSUE THREAD

**Issue by makramkd** [purescript/purescript#2775]
Hi all,

After running a `pulp init` in an empty directory and picking the 3.x releases when prompted, a `pulp build` spits out the following error:

```
Projects/pure/tutorial $ pulp build
* Building project in /Users/makram/Projects/pure/tutorial
Compiling Data.NaturalTransformation
Compiling Data.Boolean
Compiling Control.Semigroupoid
Compiling Data.Show
Compiling Control.Category
Compiling Data.Unit
Compiling Data.Void
Compiling Data.Function
Compiling Data.HeytingAlgebra
Compiling Data.Semiring
Compiling Data.Semigroup
Compiling Data.Eq
Compiling Data.Functor
Compiling Data.Ring
Compiling Data.BooleanAlgebra
Compiling Data.Ordering
Compiling Control.Apply
Compiling Data.CommutativeRing
Compiling Data.Ord.Unsafe
Compiling Data.EuclideanRing
Compiling Data.Ord
Compiling Control.Applicative
Compiling Data.Field
Compiling Control.Bind
Compiling Control.Monad
Compiling Data.Bounded
Compiling Control.Monad.Eff
Compiling Prelude
Error found:
in module Control.Monad.Eff
at /Users/makram/Projects/pure/tutorial/bower_components/purescript-eff/src/Control/Monad/Eff.purs line 36, column 1 - line 36, column 52

  Conflicting definitions are in scope for kind Effect from the following modules:

    Control.Monad.Eff
    Prim



See https://github.com/purescript/documentation/blob/master/errors/ScopeConflict.md for more information,
or to contribute content related to this error.


* ERROR: Subcommand terminated with exit code 1
```

Steps to reproduce:
1.  Create an empty directory: `mkdir tutorialv3; cd tutorialv3`
2.  Initialize an empty project: `pulp init`. Make sure you pick the "3.x" versions of the Purescript modules when prompted by the tool:
```
Unable to find a suitable version for purescript-prelude, please choose one by typing one of the numbers below:
    1) purescript-prelude#^2.1.0 which resolved to 2.5.0 and is required by purescript-eff#2.0.0
    2) purescript-prelude#^3.0.0 which resolved to 3.0.0 and is required by purescript-eff#3.0.0, tutorialv3

Prefix the choice with ! to persist it to bower.json

? Answer 2

Unable to find a suitable version for purescript-console, please choose one by typing one of the numbers below:
    1) purescript-console#^2.0.0 which resolved to 2.0.0 and is required by purescript-psci-support#2.0.0
    2) purescript-console#^3.0.0 which resolved to 3.0.0 and is required by tutorialv3

Prefix the choice with ! to persist it to bower.json

? Answer 2

Unable to find a suitable version for purescript-eff, please choose one by typing one of the numbers below:
    1) purescript-eff#^2.0.0 which resolved to 2.0.0 and is required by purescript-console#2.0.0
    2) purescript-eff#^3.0.0 which resolved to 3.0.0 and is required by purescript-console#3.0.0

Prefix the choice with ! to persist it to bower.json

? Answer 2
bower purescript-psci-support#^2.0.0     install purescript-psci-support#2.0.0

purescript-psci-support#2.0.0 bower_components/purescript-psci-support
└── purescript-console#3.0.0
```
3. Run `pulp build`. The command will fail with the message stated above.

Let me know if I am in error so that I can close this.

**Comment by paf31** [comment]
You've found us in the middle of updating libraries for the latest release, 0.11.0. This is always a slightly awkward time as we transition. You have a few options:

- Use the package versions for the 0.10.7 compiler, which basically means all but the very latest versions of the core libraries (recommended).
- Use 0.11.0 with the latest package versions (this will only work if you are only using core libraries, the ones which have been updated, right now)

You might also consider using `psc-package` which will pin versions for you, based on your compiler version.

**Comment by makramkd** [comment]
👍

**Comment by liuxiaoworld** [comment]
Hi @paf31 ,

I just did what you told, downloaded the 10.7 version of purescript and installed with stack. Then when I try to "pulp build" I got error like "can not find purs". Then I just copied psc as purs, and it passed the build and runs well. but when I try to use "pulp psci", I got the similar error like @makramkd reported, which tells me there are conflicting definitions for "kind Effect" from "Control.Monad.Eff" and "Prim".
Am I doing something wrong here?
